src/main/public/js/postcode-lookup.js

```diff
@@ -0,0 +1,158 @@
+            .classList.contains('js-hidden')
+          console.log('address', addressInputsVisible)
+          postcodeLookupWidget.querySelector('.postcode-address-visible').value = addressInputsVisible
+
+
+          var addressPickerVisible = !postcodeLookupWidget.querySelector('.postcode-address-picker')
+            .classList.contains('hidden')
+          console.log('picker', addressPickerVisible)
+          postcodeLookupWidget.querySelector('.address-selector-visible').value = addressPickerVisible
+        })
+    })
+
+})
+
+function isNorthernIrelandPostcode (postcode) {
+  return postcode.startsWith('BT')
+}
+
+function showAddressEntry (postcodeLookupWidget) {
+  postcodeLookupWidget.querySelector('.address').classList.remove('js-hidden')
+}
+
+function clearPostcodeDropdown (postcodeLookupWidget) {
+  postcodeLookupWidget.querySelector('select').innerHTML = ''
+}
+
+function hidePostcodeDropdown (postcodeLookupWidget) {
+  postcodeLookupWidget.querySelector('.postcode-address-picker').classList.add('hidden')
+}
+
+function showPostcodeDropdown (postcodeLookupWidget) {
+  postcodeLookupWidget.querySelector('.postcode-address-picker').classList.remove('hidden')
+}
+
+function clearAddressFields (postcodeLookupWidget) {
+  var addressSelector = postcodeLookupWidget.querySelector('.address')
+  addressSelector.querySelector('.address-line1').value = ''
+  addressSelector.querySelector('.address-line2').value = ''
+  addressSelector.querySelector('.address-town-or-city').value = ''
+  addressSelector.querySelector('.postcode').value = ''
+}
+
+function isAnyAddressFieldPopulated(postcodeLookupWidget) {
+  var addressSelector = postcodeLookupWidget.querySelector('.address')
+
+  return addressSelector.querySelector('.address-line1').value !== '' ||
+    addressSelector.querySelector('.address-line2').value !== '' ||
+    addressSelector.querySelector('.address-town-or-city').value !== '' ||
+    addressSelector.querySelector('.postcode').value !== ''
+}
+
+function showAddressError(isNorthernIrelandPostcode, postcodeLookupWidget) {
+  var northernIrelandErrorMessage = postcodeLookupWidget.querySelector('.postcode-search-error-ni')
+  var genericErrorMessage = postcodeLookupWidget.querySelector('.postcode-search-error')
+
+  if (isNorthernIrelandPostcode) {
+    northernIrelandErrorMessage.classList.remove('hidden')
+    genericErrorMessage.classList.add('hidden')
+  } else {
+    northernIrelandErrorMessage.classList.add('hidden')
+    genericErrorMessage.classList.remove('hidden')
+  }
+
+  postcodeLookupWidget.querySelector('.postcode-search-container').classList.add('form-group-error')
+}
+
+function lookupPostcode (postcode, postcodeLookupWidget) {
+  var xhr = new XMLHttpRequest()
+  xhr.open('GET', '/postcode-lookup?postcode=' + encodeURIComponent(postcode))
+  xhr.onload = function () {
+    if (xhr.status !== 200) {
+      showAddressError(false, postcodeLookupWidget)
+      showAddressEntry(postcodeLookupWidget)
```

Comment:
What do you think about adding `show(element)` and `hide(element)` functions? Then, combined with my other comment on this file` we could have e.g. `show(postcodeLookupWidget())`.

Reply:
showing and hiding is a bit specific depending on how its hidden because we have to support a non js fall back, there's `js-hidden` `hidden` and `js-visible` not sure if it could be made generic

Reply:
Don't bother if it would overcomplicate the code.